src/components/MobileMenu/MobileMenu.tsx

```diff
@@ -0,0 +1,22 @@
+import "./mobile-menu.sass";
+import ProductCategoryMenu from "../ProductCategoryMenu/ProductCategoryMenu";
+
+interface Props {
+  modalIsActive: number;
```

Comment:
Exactly!  That prop is only meant to control whether the modal is visible.  The reason I'm using a number  instead of a boolean is because I only want one of the modals (either the MobileMenu or CartModal) to be open at a time.  If they are both open, they crash into each other and cause visual problems. I got this idea from a React developer exercise inside their docs where only one component should display at a time.  Could you suggest a different way of making it so that only the MobileMenu or Cart could be unhidden at a time?  Thank you for all of this!!!

Reply:
ahh gotcha! your solution isn't bad. I would still make this prop a boolean and do the number logic in the parent component though:

```tsx
      <MobileMenu
        isOpen={modalIsActive === 1}
        handleMenuClick={handleMenuClick}
      />
      <CartModal
        isOpen={modalIsActive === 2}
        handleCartClick={handleCartClick}
      />
```

and maybe rename `modalIsActive` to `modalVisibilityIndicator`

the benefit of this is that your modal components remain readable and more reusable. if you really want to fully guarantee that the modals don't clash, you could add an `isMobileMenuOpen` and an `isCartModalOpen` in your global state, and then inside each modal, check the visibility of the other modal